bindings/java/src/junit/com/apple/foundationdb/tuple/TupleEncodingTest.java

```diff
@@ -0,0 +1,526 @@
+/*
+ * TupleEncodingTest.java
+ *
+ * This source file is part of the FoundationDB open source project
+ *
+ * Copyright 2013-2018 Apple Inc. and the FoundationDB project authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.apple.foundationdb.tuple;
+
+import java.lang.reflect.Field;
+import java.math.BigInteger;
+import java.nio.ByteBuffer;
+import java.nio.ByteOrder;
+import java.security.AccessController;
+import java.security.PrivilegedExceptionAction;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.UUID;
+import java.util.stream.Stream;
+
+import com.apple.foundationdb.FDB;
+import com.apple.foundationdb.FDBLibraryRule;
+import com.apple.foundationdb.subspace.Subspace;
+
+import org.hamcrest.MatcherAssert;
+import org.hamcrest.Matchers;
+import org.junit.Assert;
+import org.junit.Assume;
+import org.junit.ClassRule;
+import org.junit.Test;
+import org.junit.experimental.theories.DataPoints;
+import org.junit.experimental.theories.FromDataPoints;
+import org.junit.experimental.theories.Theories;
+import org.junit.experimental.theories.Theory;
+import org.junit.runner.RunWith;
+
+/**
+ * Tests around packing, versionstamps, and assorted encoding-related stuff for tuples.
+ */
+@RunWith(Theories.class)
+public class TupleEncodingTest {
+    private static final byte FF = (byte) 0xff;
+    @ClassRule
+    public static final FDBLibraryRule fdbLib = FDBLibraryRule.v7();
+
+    @DataPoints("baseTuples")
+    public static final List<Tuple> baseTuples = Arrays.asList(new Tuple(), Tuple.from(), Tuple.from((Object) null),
+            Tuple.from("prefix"), Tuple.from("prefix", null), Tuple.from(new UUID(100, 1000)),
+            Tuple.from(Versionstamp.incomplete(1)), Tuple.from(Tuple.from(Versionstamp.incomplete(2))),
+            Tuple.from(Collections.singletonList(Versionstamp.incomplete(3))));
+
+    @DataPoints("addItems")
+    public static final List<Object> items = Arrays.asList(null, 1066L, BigInteger.valueOf(1066), -3.14f, 2.71828,
```

Comment:
Should there be a _big_ `BigInteger` in here?

Reply:
Was inherited from a previous test implementation, but it's not like it's hard to add, so sure :)